crypto/src/main/java/org/web3j/crypto/transaction/type/Transaction4844.java

```diff
@@ -0,0 +1,293 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.web3j.crypto.transaction.type;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.web3j.abi.datatypes.Bytes;
+import org.web3j.crypto.Blob;
+import org.web3j.crypto.BlobUtils;
+import org.web3j.crypto.Sign;
+import org.web3j.rlp.RlpList;
+import org.web3j.rlp.RlpString;
+import org.web3j.rlp.RlpType;
+import org.web3j.utils.Numeric;
+
+public class Transaction4844 extends Transaction1559 implements ITransaction {
+
+    private final BigInteger maxFeePerBlobGas;
+    private final List<Bytes> versionedHashes;
+    private final Optional<List<Blob>> blobs;
+    private final Optional<List<Bytes>> kzgProofs;
+    private final Optional<List<Bytes>> kzgCommitments;
+
+    public Transaction4844(
+            long chainId,
+            BigInteger nonce,
+            BigInteger maxPriorityFeePerGas,
+            BigInteger maxFeePerGas,
+            BigInteger gasLimit,
+            String to,
+            BigInteger value,
+            String data,
+            BigInteger maxFeePerBlobGas,
+            List<Bytes> versionedHashes) {
+        super(chainId, nonce, gasLimit, to, value, data, maxPriorityFeePerGas, maxFeePerGas);
+        this.maxFeePerBlobGas = maxFeePerBlobGas;
+        this.versionedHashes = versionedHashes;
+        this.blobs = Optional.empty();
+        this.kzgCommitments = Optional.empty();
+        this.kzgProofs = Optional.empty();
+    }
+
+    public Transaction4844(
+            List<Blob> blobs,
+            List<Bytes> kzgCommitments,
+            List<Bytes> kzgProofs,
+            long chainId,
+            BigInteger nonce,
+            BigInteger maxPriorityFeePerGas,
+            BigInteger maxFeePerGas,
+            BigInteger gasLimit,
+            String to,
+            BigInteger value,
+            String data,
+            BigInteger maxFeePerBlobGas,
+            List<Bytes> versionedHashes) {
+        super(chainId, nonce, gasLimit, to, value, data, maxPriorityFeePerGas, maxFeePerGas);
+        this.maxFeePerBlobGas = maxFeePerBlobGas;
+        this.versionedHashes = versionedHashes;
+        this.blobs = Optional.ofNullable(blobs);
+        this.kzgCommitments = Optional.ofNullable(kzgCommitments);
+        this.kzgProofs = Optional.ofNullable(kzgProofs);
+    }
+
+    public Transaction4844(
+            List<Blob> blobsData,
```

Comment:
I see here it looks like we have a method of calculating the proofs and related based on the blob data, which was what my earlier question/comment was about. I think that makes sense, so just wanted to confirm my understanding.